### PR TITLE
Adds feature to search users by username.

### PR DESCRIPTION
**Pull request by Jesus**
Because:

- It's a requested feature!

This commit:

- Adds the full feature (form in views, action in controller, etc.)
- Adds feature tests.

### CODE REVIEW

File: spec/features/search_by_username_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  scenario "ignoring not matching users" do
+    given_we_have_these_users %w(anthony patrick jesus matthew daniel)
+    and_we_search_the_term "thony"
+    then_the_page_should_not_contain_user "patrick"
+  end
+
+  def given_we_have_these_users(usernames)
+    usernames.each do |username|
+      create :user,
+        username: username,
```
Comment (Jesus) [inline spec/features/search_by_username_spec.rb]: seriously?

Comment (thedanotto) [inline spec/features/search_by_username_spec.rb]: @Jesus Jeez, looks like you've been caught red handed again ;)

**Comment by thedanotto** [comment]
I pulled down your changes @Jesus and fixed the merge conflict in #17.

**Comment by Jesus** [comment]
Thanks for your comments!

Pagination is a good idea, the only reason why I'm not adding yet is because we have a tight timeframe at this stage. But I'll ticket that so we can add it later if we get the time.

I'm not sure how that resourceful route you're suggesting could work. If you have an example I'd love to see it.

Thanks also for merging the changes =)

**Comment by thedanotto** [comment]
Completely agree on all fronts. Time to stop sweating the small stuff and get some of the major features implemented. Bed time for now. Haha

**Pull request closed without merging**